compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -642,39 +669,12 @@ def IREELinalgExt_AttentionOp : IREELinalgExt_Op<"attention",
       return sumType->getRank();
     }
 
-    // Method to implement for specifying output range for
-    // DestinationStyleOpInterface
-    MutableOperandRange getDpsInitsMutable() {
-      return getOutputsMutable();
-    }
-
-    SmallVector<AffineMap> getIndexingMapsArray();
-
-    AffineMap getQueryMap() {
-      return cast<AffineMap>(getIndexingMapsArray()[0]);
-    }
-    AffineMap getKeyMap() {
-      return cast<AffineMap>(getIndexingMapsArray()[1]);
-    }
-    AffineMap getValueMap() {
-      return cast<AffineMap>(getIndexingMapsArray()[2]);
-    }
-    AffineMap getOutputMap() {
-      return cast<AffineMap>(getIndexingMapsArray()[3]);
-    }
-    std::optional<AffineMap> getMaxMap() {
-      if (getNumResults() < 2)
-        return std::nullopt;
-      return cast<AffineMap>(getIndexingMapsArray()[4]);
-    }
-    std::optional<AffineMap> getSumMap() {
-      if (getNumResults() < 3)
-        return std::nullopt;
-      return cast<AffineMap>(getIndexingMapsArray()[5]);
-    }
-
-    int64_t getIterationDomainRank() {
-      return getQueryMap().getNumDims();
+    // Use indexing_map to infer if V is transposed.
+    bool getTransposeV() {
+      AffineMap vMap = getValueMap();
+      auto vFastDim = llvm::cast<AffineDimExpr>(vMap.getResults().back());
+      int64_t nDim = vMap.getNumInputs() - 1;
+      return nDim != vFastDim.getPosition();
```

Comment:
Oh yeah, youre right! :)

Reply:
@hanhanW actually turns out for minorIdentity would work in the base case of 
(m, k1, k2, n) -> (k2, n) since it needs to be contiguous identity dims, but for most part we'd have (b0, m, k1, k2, n) -> (b0, k2, n). Making it a non-minor identity.

Reply:
I also tried projectedPermutation and minorIdentityWithBroadcast both which also doesn't cover some cases haha.

Reply:
oh I see, I forgot that there could be batch dimensions. Sorry for the bad suggestion and thanks for the updates in the comments.

Reply:
haha nw! I was excited for that clean up too :D